src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,208 @@
+        {
+            auto const result{m_deserializer->deserialize_log_event()};
+            if (result.has_error())
+            {
+                auto const error{result.error()};
+                if (std::errc::result_out_of_range == error)
+                {
+                    std::cerr << "File contains an incomplete IR stream" << std::endl;
+                }
+                else if (std::errc::no_message_available != error)
+                {
+                    std::cerr << "Failed to decompress: "
+                              << error.category().name() << ":" << error.message()
+                              << std::endl;
+                }
+                break;
+            }
+
+            numDecoded++;
+
+            m_log_events.push_back(result.value());
+        }
+
+        return numDecoded;
+    }
+
+    bool decode(const emscripten::val &results, size_t startIdx, size_t endIdx)
+    {
+        for (size_t log_event_idx = startIdx; log_event_idx < endIdx; log_event_idx++)
+        {
+            auto const &log_event{m_log_events[log_event_idx]};
+            std::string message;
+
+            {
+                auto constant_handler = [&](std::string const &value, size_t begin_pos, size_t length)
+                {
+                    message.append(value, begin_pos, length);
+                };
+
+                auto encoded_int_handler = [&](clp::ir::four_byte_encoded_variable_t value)
+                { message.append(clp::ffi::decode_integer_var(value)); };
+
+                auto encoded_float_handler = [&](clp::ir::four_byte_encoded_variable_t encoded_float)
+                {
+                    message.append(clp::ffi::decode_float_var(encoded_float));
+                };
+
+                auto dict_var_handler = [&](std::string const &dict_var)
+                { message.append(dict_var); };
+
+                try
+                {
+                    clp::ffi::ir_stream::generic_decode_message<true>(
+                        log_event.get_logtype(),
+                        log_event.get_encoded_vars(),
+                        log_event.get_dict_vars(),
+                        constant_handler,
+                        encoded_int_handler,
+                        encoded_float_handler,
+                        dict_var_handler);
+                }
+                catch (clp::ffi::ir_stream::DecodingException const &e)
+                {
+                    std::cerr << "Failed to decode. Error Code: " << static_cast<uint32_t>(clp::ffi::ir_stream::IRErrorCode_Decode_Error)
+                              << std::endl;
+
+                    break;
+                }
+            }
+
+            LOG_VERBOSITY verbosity_level = LOG_VERBOSITY::NONE;
+            for (size_t verbosity_idx = 1; verbosity_idx < sizeof(VERBOSITIES) / sizeof(VERBOSITIES[0]); verbosity_idx++)
+            {
+                if (log_event.get_logtype().substr(1).starts_with(VERBOSITIES[verbosity_idx]))
```

Comment:
it might make more sense to compare against the decoded message. The log level might be a dictionary variable and thus not in the logtype.